src/coreclr/src/gc/gc.cpp

```diff
@@ -18090,7 +18090,7 @@ uint8_t* gc_heap::find_object (uint8_t* interior)
         heap_segment* seg = find_segment (interior, FALSE);
         if (seg
 #ifdef FEATURE_CONSERVATIVE_GC
-            && (GCConfig::GetConservativeGC() || interior <= heap_segment_allocated(seg))
+            && (!GCConfig::GetConservativeGC() || interior <= heap_segment_allocated(seg))
```

Comment:
Note that this effects the path we take here in the default CoreCLR config. CoreCLR has `FEATURE_CONSERVATIVE_GC` defined and `GCConfig::GetConservativeGC` defaults to false.

It means that this was effectively `if (seg && interior <=  heap_segment_allocated(seg))` before this change in the default CoreCLR config, and it is just `if (seg)` after this change.

Reply:
these checks for `GetConservativeGC` seem suspicious to me - I dunno who added them but the impl seems questionable... the idea, if I had to guess, was probably that if we are in conservative mode (ie the config is set), we can tolerate an interior pointer that's > heap_segment_allocated but all we do is we return 0 there - the only case where we could return a non zero value is if an interior pointer is >= heap_segment_mem and < heap_segment_allocated, regardless of whether the conservative config is set. so the code should just be 

```cplusplus
        if (seg && (interior < heap_segment_allocated(seg))
        {
```
and the assert can be deleted.

Reply:
if we want to be more complete, we could do 
```cplusplus
#ifdef FEATURE_CONSERVATIVE_GC
            if (interior >= heap_segment_allocated (seg))
                return 0;
#else
            assert (interior < heap_segment_allocated (seg));
#endif
```
meaning that if we find a valid seg for it, we tolerate it if conservative config is set and inteiror is > heap_segment_allocated.

Reply:
Thanks for the comments.  I believe I have updated as per the last comment.  I tried the Wasm Generics test in a loop 100 times and it completed without errors, so this fix looks good for CoreRT/Wasm.  I'm sure you experts will advise if I have it wrong.